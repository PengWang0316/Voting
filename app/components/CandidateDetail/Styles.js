import { StyleSheet } from 'react-native';

import Theme from '../../Theme';

export default StyleSheet.create({
  root: {
    width: '100%',
    minHeight: '100%',
  },
  interView: {
    flex: 1,
    alignItems: 'center',
    paddingTop: 50,
    paddingBottom: 30,
    paddingLeft: 40,
    paddingRight: 40,
  },
  democracyBg: {
    backgroundColor: '#1a65dd',
  },
  republicBg: {
    backgroundColor: '#dd201a',
  },
  nameText: {
    fontSize: 35,
    fontWeight: 'bold',
    color: Theme.primary.contrastText,
    marginTop: 30,
    marginBottom: 30,
  },
  descriptionText: {
    fontSize: 20,
    color: Theme.primary.contrastText,
  },
  avatar: {
    width: 200,
    height: 200,
    borderRadius: 100,
  },
  voteBtn: {
    backgroundColor: Theme.primary.main,
    borderRadius: 15,
    marginTop: 35,
    width: '100%',
  },
});
