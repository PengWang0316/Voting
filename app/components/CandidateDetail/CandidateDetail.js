import React from 'react';
import { View, Text, ScrollView } from 'react-native';
import CacheableImage from 'react-native-cacheable-image';
import { Button, Icon } from 'react-native-elements';
import { connect } from 'react-redux';

import Styles from './Styles';

type Props = {
  candidate: Object,
  user: Object,
};

export const CandidateDetail = ({ candidate, user }: Props) => (
  <View style={[Styles.root, candidate.isDemocracy ? Styles.democracyBg : Styles.republicBg]}>
    <ScrollView>
      <View style={Styles.interView}>
        <CacheableImage
          style={Styles.avatar}
          source={{ uri: candidate.photo }}
        />
        <Text style={Styles.nameText}>{candidate.name}</Text>
        <Text style={Styles.descriptionText}>{candidate.description}</Text>
        <Button
          icon={(
            <Icon
              name="vote"
              type="material-community"
              color="white"
            />
          )}
          title="VOTE FOR ME"
          buttonStyle={Styles.voteBtn}
          disabled={!user.id}
        />
      </View>
    </ScrollView>
  </View>
);
/* istanbul ignore next */
const mapStateToProps = state => ({
  user: state.user,
});
export default connect(mapStateToProps, null)(CandidateDetail);
