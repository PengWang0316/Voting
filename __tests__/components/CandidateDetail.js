import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { CandidateDetail } from '../../app/components/CandidateDetail/CandidateDetail';

jest.mock('react-native-elements', () => ({ Button: 'Button', Icon: 'Icon' }));
jest.mock('react-native-cacheable-image', () => 'CacheableImage');

describe('CandidateDetail', () => {
  const defaultProps = {
    candidate: {
      photo: 'photoURI',
      name: 'name',
      description: 'description',
    },
    user: {},
  };

  test('Snapshot without user id', () => expect(renderer.create(<CandidateDetail {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('Snapshot with user id', () => expect(renderer.create(<CandidateDetail {...{ ...defaultProps, user: { id: 'id' } }} />).toJSON()).toMatchSnapshot());
});
