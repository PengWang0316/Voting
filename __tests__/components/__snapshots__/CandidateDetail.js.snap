// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidateDetail Snapshot with user id 1`] = `
<View
  style={
    Array [
      Object {
        "minHeight": "100%",
        "width": "100%",
      },
      Object {
        "backgroundColor": "#dd201a",
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "paddingBottom": 30,
            "paddingLeft": 40,
            "paddingRight": 40,
            "paddingTop": 50,
          }
        }
      >
        <CacheableImage
          source={
            Object {
              "uri": "photoURI",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "height": 200,
              "width": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 35,
              "fontWeight": "bold",
              "marginBottom": 30,
              "marginTop": 30,
            }
          }
        >
          name
        </Text>
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          description
        </Text>
        <Button
          buttonStyle={
            Object {
              "backgroundColor": "#81c784",
              "borderRadius": 15,
              "marginTop": 35,
              "width": "100%",
            }
          }
          disabled={false}
          icon={
            <Icon
              color="white"
              name="vote"
              type="material-community"
            />
          }
          title="VOTE FOR ME"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`CandidateDetail Snapshot without user id 1`] = `
<View
  style={
    Array [
      Object {
        "minHeight": "100%",
        "width": "100%",
      },
      Object {
        "backgroundColor": "#dd201a",
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "paddingBottom": 30,
            "paddingLeft": 40,
            "paddingRight": 40,
            "paddingTop": 50,
          }
        }
      >
        <CacheableImage
          source={
            Object {
              "uri": "photoURI",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "height": 200,
              "width": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 35,
              "fontWeight": "bold",
              "marginBottom": 30,
              "marginTop": 30,
            }
          }
        >
          name
        </Text>
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          description
        </Text>
        <Button
          buttonStyle={
            Object {
              "backgroundColor": "#81c784",
              "borderRadius": 15,
              "marginTop": 35,
              "width": "100%",
            }
          }
          disabled={true}
          icon={
            <Icon
              color="white"
              name="vote"
              type="material-community"
            />
          }
          title="VOTE FOR ME"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
